### PR TITLE
renaming and moving functional tests and setting up unit tests template framework

### PR DESCRIPTION
**Pull request by screeley44**

### CODE REVIEW

File: pkg/importer/dataStream_test.go [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+import (
+	"io"
+	"net/url"
+	"reflect"
+	"testing"
+
+	. "github.com/kubevirt/containerized-data-importer/pkg/common"
+)
+
+func Test_dataStream_Read(t *testing.T) {
```
Comment (jeffvance) [inline pkg/importer/dataStream_test.go]: Why `Test_` here vs. `TestXyz` elsewhere?

Comment (screeley44) [reply pkg/importer/dataStream_test.go]: I think so

File: test/functional/controller/controller_suite_test.go [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // +build unit_test
```
Comment (jeffvance) [inline test/functional/controller/controller_suite_test.go]: is this still applicable after the test has moved to _functional/_?

File: test/functional/controller/controller_test.go [hunk 3]
```diff
@@ -0,0 +1,291 @@
+// +build unit_test
```
Comment (jeffvance) [inline test/functional/controller/controller_test.go]: ditto

File: test/framework/fileConversion.go [hunk 4]
```diff
@@ -19,29 +19,26 @@ var formatTable = map[string]func(string) (string, error){
 
 func FormatTestData(srcFile string, targetFormats ...string) (string, error) {
 
-	var (
-		err     error
-		outFile = srcFile
-	)
+	outFile := srcFile
```
Comment (jeffvance) [inline test/framework/fileConversion.go]: In my diff I added a comment to this func which you may wish to keep:
```
 // create file based on targetFormat extensions and return created file's name.
// note: intermediate files are removed.

```

File: test/functional/controller/controller_suite_test.go (hunk 2 above)
Comment (copejon) [reply test/functional/controller/controller_suite_test.go]: Probably not,  they were useful early on but all build tags should probably be dropped until we come up with a standard for them.

**Pull request merged**